Include/Rocket/Core/StringBase.inl

```diff
@@ -26,42 +26,42 @@
  */
 
 template< typename T >
-StringBase< T >::StringBase() : value((T*)local_buffer), buffer_size(LOCAL_BUFFER_SIZE), length(0), hash(0)
+StringBase< T >::StringBase() : value(LocalBuffer()), buffer_size(LOCAL_BUFFER_SIZE), length(0), hash(0)
```

Comment:
What complexity? It's a inline function that's part of a template anyway that folds to the same code as before. So there is no runtime complexity?

Compile time complexity? It's simpler to read in my view because casts usually indicate there is something I need to be worried about. That's a distraction. Plus if you do change the underlying type for some reason, you could. That's a cost free abstraction to my mind. But if you think it's clearer to litter the code with casts and lose that. Keep what you had. It's a design choice I wouldn't make, but it's no big deal so if that's your preference by all means stick with it.